### ISSUE TITLE
Very inconsistent performance

### ISSUE THREAD

**Issue by beew** [sympy/sympy#25981]
Hi, I ran this same code several times (in the same python session) and the run time varies drastically 

```
import time

from sympy import primefactors

```
The below part was run repeatedly

```
start = time.time(); primefactors(1111111111111111111111111111111111111111111111111111111111111111111111111111111111); time.time() - start

```
While the actual factorization is the same for each run, the run time varies over a huge range. In order:

> 153s, 3.25s, 6.70s, 27.8s, 48.0s, 26.1s, 96.4s, 147s,  17.4s, 248s, 50s, 32.5s, 145s

symp1.12, python3.10.9, Ubuntu 22.04.

**Comment by asmeurer** [comment]
Maybe @haru-44 can confirm, this was likely fixed by one of his PRs.

**Comment by asmeurer** [comment]
I tried some bigger numbers and the timing in master seems to always be consistent. I think whatever the issue was here has been fixed.

**Comment by beew** [comment]
Yes. I tested with master and the run above is ~ 14s always as oppose to ranging from 4s to 250s for sympy1.12

**Comment by haru-44** [comment]
There was a similar issue (#25253) before and it has already been fixed.
It will be resolved in sympy 1.13.
